### PR TITLE
Update text input maxlength to check for custom values first

### PR DESCRIPTION
**Pull request by sarah-wfaa**

### CODE REVIEW

File: src/components/Input/index.js [hunk 1]
```diff
@@ -171,23 +171,23 @@ const Input = ({ fieldData, name, value, fieldHidden, subfield, fromNameField, .
...
-                maxLength={type === 'phone' ? 26 : type === 'text' ? 256 : maxLength || 524288} // 524288 = 512kb, avoids invalid prop type error if maxLength is undefined.
+                maxLength={maxLength ? maxLength : type === 'phone' ? 26 : type === 'text' ? 256 : 524288} // 524288 = 512kb, avoids invalid prop type error if maxLength is undefined.
                 name={name}
                 placeholder={placeholder}
                 {...register(name, {
                     value: defaultValue,
                     required: !fieldHidden ? isRequired && strings.errors.required : false,
-                    maxLength: type === 'phone' ? {
+                    maxLength: maxLength > 0 && maxLength ? {
+                        value: maxLength > 0 && maxLength,
```
Comment (lee-wfaa) [inline src/components/Input/index.js]: On line 182 there is a second check on maxLength > 0 and if maxLength.

File: src/components/Input/index.js [hunk 2]
```diff
@@ -171,23 +171,23 @@ const Input = ({ fieldData, name, value, fieldHidden, subfield, fromNameField, .
...
+                        message:
+                            maxLength > 0 &&
```
Comment (lee-wfaa) [inline src/components/Input/index.js]: Same here: maxLength is greater than 0 if this line of code runs.

**Review by lee-wfaa — APPROVED** [review]
Looks good!

File: src/components/Input/index.js (hunk 2 above)
Comment (sarah-wfaa) [reply src/components/Input/index.js]: updated, thanks!!

**Review by lee-wfaa — COMMENTED** [review]
```javascript
maxLength: fromNameField
  ? {
      value: 50,
      message: 'Name must be less than 50 characters.',
    }
  : maxLength > 0
  ? {
      value: maxLength,
      message: `${strings.errors.maxChar.front}  ${maxLength} ${strings.errors.maxChar.back}`,
    }
  : undefined
```
In this version, maxLength is checked only once to see if it's greater than 0. If it is, it sets the value and message accordingly. If it's not, it sets the value to undefined (which is only preference since it is like maxLength was never passed as a prop (but null works fine too if maxLength can handle it in the react-forms library).

Also does this override the initial maxLength prop?

**Pull request merged**